src/probnum/_config.py

```diff
@@ -0,0 +1,83 @@
+import contextlib
+from typing import Any
+
+
+class Configuration:
+    """
+    Configuration over which some mechanics of probnum can be controlled dynamically.
+
+    ``probnum`` provides some configurations together with default values. These
```

Comment:
Super minor: I think "ProbNum" instead of "probnum" is more consistent with the rest of the package here (http://www.probabilistic-numerics.org/en/latest/development/styleguide.html)

Reply:
Changed it :)